dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/CompositionHost.cs

```diff
@@ -31,42 +32,29 @@
 
 namespace VisualLayerIntegration
 {
-      class CompositionHost : HwndHost
+    sealed class CompositionHost : HwndHost
     {
-        object dispatcherQueue;
-        int hostHeight, hostWidth;
-        double hostDpiX, hostDpiY;
-        private ICompositionTarget compositionTarget;
-        internal const int
-            WS_CHILD = 0x40000000,
-            WS_VISIBLE = 0x10000000,
-            LBS_NOTIFY = 0x00000001,
-            HOST_ID = 0x00000002,
-            LISTBOX_ID = 0x00000001,
-            WS_VSCROLL = 0x00200000,
-            WS_BORDER = 0x00800000;
-
-
-        public Visual Child
-        {
-            set
-            {
-                if (Compositor == null)
-                {
-                    InitComposition(hwndHost);
-                }
-                compositionTarget.Root = value;
-            }
-        }
+        private object _dispatcherQueue;
+        private ICompositionTarget _compositionTarget;
+        private List<IDisposable> registeredDisposables = new List<IDisposable>();
+       
         public IntPtr hwndHost { get; private set; }
```

Comment:
>blic IntPtr hwnd [](start = 10, length = 16)

I don't think this is needed. It's passed to the BarGraph but it shouldn't be.

Reply:
Access to this is needed in BarGraphHostControl to set properties to create the sharpdx render target 

---
In reply to: [276430589](https://github.com/Microsoft/Windows.UI.Composition-Win32-Samples/pull/1#discussion_r276430589) [](ancestors = 276430589)